controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/SealStreamTask.java

```diff
@@ -72,6 +91,43 @@ public SealStreamTask(final StreamMetadataTasks streamMetadataTasks,
                 });
     }
 
+    private CompletableFuture<Boolean> cancelTransactions(OperationContext context, String scope, String stream) {
+        return streamMetadataStore.getActiveTxns(scope, stream, context, executor)
+                .thenCompose(activeTxns -> {
+                    if (activeTxns == null || activeTxns.isEmpty()) {
+                        return CompletableFuture.completedFuture(true);
+                    } else {
+                        // abort transactions
+                        return Futures.allOfWithResults(activeTxns.entrySet().stream().map(txIdPair -> {
```

Comment:
This boils down to `Futures.allOfWithResults(List<CompletableFuture<Void>>) .`
Q: If we are using the results can we just use `CompletableFuture.allOf(futuresOfactiveTxnCancellation.toArray())` directly?
Q: Should'nt we return false only if any abortTxn fails. (i.e. If we abort all txns successfully then why return false?)

Reply:
>  Q: Should'nt we return false only if any abortTxn fails. (i.e. If we abort all txns successfully then why return false?)

No, the abortTxn is performed asynchronously on abortEventProcessor. Which means this method call simply sets txn state to aborting and posts an event in the abort stream.. 
So here we basically initiate abort txn and then return requesting postponement of processing of this event.. when this event is picked for processing again, this check repeats until there are no active txns left on the stream.